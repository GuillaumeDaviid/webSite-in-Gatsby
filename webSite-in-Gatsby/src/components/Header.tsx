import React, { useState } from 'react'
import './styles/Header.scss'

export default function Header() {
    const [visible, setVisible] = useState(false);

    function handleClick() {
        setVisible(!visible)
    }


    return(
        <section id="home">
        <div className="Header">

        <nav className={`Header_nav ${visible ? `nav_visible` : `nav_notVisible`}`} data-testid="nav">
                <ul>
                <li data-testid="li"><a href="#home" onClick={handleClick}><span>Accueil</span></a></li>
                <li data-testid="li"><a href="#portfolio" onClick={handleClick}><span>Portfolio</span></a></li>
                <li data-testid="li"><a href="#contact" onClick={handleClick}><span>Contact</span></a></li>
                <li data-testid="li"><a href="/blog" onClick={handleClick}><span>Blog</span></a></li>
                </ul>
                <a href="#" className="cd-close-menu" onClick={handleClick} data-testid="closeNav">Close<span></span></a>
            </nav>

            <p className="cd-menu-trigger" onClick={handleClick} data-testid="burger-menu"><span></span></p>
            <h1 className="Header_title" data-testid="title">GUILLAUME DAVID</h1>

            <div className="Header_separator-white"></div>

            <h2 className="Header_description" data-testid="subTitle">DÃ©veloppeur Front End / Javascript / React</h2>


            <a className="mouse-scroll" href="#introduction" data-testid="scroll">
						<span className="mouse">
							<span className="mouse-movement"></span>
						</span>
						<span className="mouse-message fadeIn">scroll</span>
			</a>
        </div>
        </section>
    )
}