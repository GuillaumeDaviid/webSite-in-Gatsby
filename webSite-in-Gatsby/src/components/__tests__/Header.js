import React from "react"
import renderer from "react-test-renderer"
import Header from "../Header.tsx"
import { render, screen } from "@testing-library/react"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Header />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should have a title element", () => {
    render(<Header />)

    const title = screen.getByTestId('title');
    expect(title).toBeTruthy();
  })

  it("should have a subtitle element", () => {
    render(<Header />)

    const subTitle = screen.getByTestId('subTitle');
    expect(subTitle).toBeTruthy();
  })

  it("should have a nav element", () => {
    render(<Header />)

    const nav = screen.getByTestId('nav');
    expect(nav).toBeTruthy();
  })

  it("should have a scroll element", () => {
    render(<Header />)

    const scroll = screen.getByTestId('scroll');
    expect(scroll).toBeTruthy();
  })

  it("should have a closeNav element", () => {
    render(<Header />)

    const closeNav = screen.getByTestId('closeNav');
    expect(closeNav).toBeTruthy();
  })

  it("should have 4 elements in nav section", () => {
    render(<Header />)

    const li = screen.getAllByTestId('li');
    expect(li).toHaveLength(4);
  })
})