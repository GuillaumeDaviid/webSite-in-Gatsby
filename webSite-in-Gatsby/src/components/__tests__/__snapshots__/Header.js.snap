// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<section
  id="home"
>
  <div
    className="Header"
  >
    <nav
      className="Header_nav nav_notVisible"
    >
      <ul>
        <li>
          <a
            href="#home"
            onClick={[Function]}
          >
            <span>
              Accueil
            </span>
          </a>
        </li>
        <li>
          <a
            href="#portfolio"
            onClick={[Function]}
          >
            <span>
              Portfolio
            </span>
          </a>
        </li>
        <li>
          <a
            href="#contact"
            onClick={[Function]}
          >
            <span>
              Contact
            </span>
          </a>
        </li>
        <li>
          <a
            href="/blog"
            onClick={[Function]}
          >
            <span>
              Blog
            </span>
          </a>
        </li>
      </ul>
      <a
        className="cd-close-menu"
        href="#"
        onClick={[Function]}
      >
        Close
        <span />
      </a>
    </nav>
    <a
      className="cd-menu-trigger"
      href="#main-nav"
      onClick={[Function]}
    >
      <span />
    </a>
    <h1
      className="Header_title"
    >
      GUILLAUME DAVID
    </h1>
    <div
      className="Header_separator-white"
    />
    <h3
      className="Header_description"
    >
      DÃ©veloppeur Front End / Javascript / React
    </h3>
    <a
      className="mouse-scroll"
      href="#introduction"
    >
      <span
        className="mouse"
      >
        <span
          className="mouse-movement"
        />
      </span>
      <span
        className="mouse-message fadeIn"
      >
        scroll
      </span>
    </a>
  </div>
</section>
`;
